src/audio/module_adapter/iadk/iadk_module_adapter.cpp

```diff
@@ -80,7 +80,14 @@ uint32_t IadkModuleAdapter::IadkModuleAdapter_Process(struct sof_source **source
 			new (&output_stream_buffers[i]) intel_adsp::OutputStreamBuffer(osb_data);
 		}
 
-		ret = processing_module_.Process(input_stream_buffers, output_stream_buffers);
+		uint32_t iadk_ret =
+			processing_module_.Process(input_stream_buffers, output_stream_buffers);
+
+		/* IADK modules returns uint32_t return code. Convert to failure if Process
+		 * not successful.
+		 */
+		if (iadk_ret != 0)
+			ret = -ENODATA;
```

Comment:
Not really for this PR, but I wonder if it's ok to sink_commit_buffer() with full "output_stream_buffers[i].size" in case of error.

Reply:
Yes it is question whether we plan to recover in case of failure in other way than just restart pipeline. 
Currently we do not recover anyway:
https://github.com/thesofproject/sof/blob/b617388c622f29250221d309c20241c4953efd41/src/audio/pipeline/pipeline-schedule.c#L380C1-L388C2